internal/reconciler/logpipeline/reconciler.go

```diff
@@ -153,54 +156,88 @@ func (r *Reconciler) doReconcile(ctx context.Context, pipeline *telemetryv1alpha
 }
 
 func (r *Reconciler) reconcileFluentBit(ctx context.Context, name types.NamespacedName, pipeline *telemetryv1alpha1.LogPipeline, checksum string) error {
-	shouldDeleteFluentBit, err := r.isLastPipelineMarkedForDeletion(ctx, pipeline)
-	if err != nil {
-		return fmt.Errorf("failed to check if LogPipeline is last marked for deletion: %v", err)
-	}
-
-	if shouldDeleteFluentBit {
-		return deleteFluentBit(ctx, r, name)
-	}
-
 	serviceAccount := commonresources.MakeServiceAccount(name)
-	if err = utils.CreateOrUpdateServiceAccount(ctx, r, serviceAccount); err != nil {
+	if err := setOwnerReference(pipeline, serviceAccount, r.Scheme()); err != nil {
```

Comment:
Let's revert this changes. We can set the owner flag here and clear it on other owners while merging the metadata.